File: backend/app/schemas/device.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class DeviceCreate(BaseModel):
    name: str
    host: str
    port: int = 443
    use_ssl: bool = True
    username: str
    password: str
    notes: Optional[str] = None


class DeviceUpdate(BaseModel):
    name: Optional[str] = None
    host: Optional[str] = None
    port: Optional[int] = None
    use_ssl: Optional[bool] = None
    username: Optional[str] = None
    password: Optional[str] = None
    notes: Optional[str] = None
    is_active: Optional[bool] = None


class DeviceResponse(BaseModel):
    id: int
    name: str
    host: str
    port: int
    use_ssl: bool
    username: str
    model: Optional[str] = None
    serial_number: Optional[str] = None
    ros_version: Optional[str] = None
    board_name: Optional[str] = None
    notes: Optional[str] = None
    is_active: bool
    last_seen: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    created_by: int

    model_config = {"from_attributes": True}


class DeviceStatus(BaseModel):
    id: int
    name: str
    host: str
    online: bool
    uptime: Optional[str] = None
    cpu_load: Optional[int] = None
    memory_used: Optional[int] = None
    memory_total: Optional[int] = None
    hdd_used: Optional[int] = None
    hdd_total: Optional[int] = None
    ros_version: Optional[str] = None
    board_name: Optional[str] = None
    architecture: Optional[str] = None

File: frontend/src/types/device.ts
export interface Device {
  id: number;
  name: string;
  client_id: number | null;
  host: string;
  api_port: number;
  use_ssl: boolean;
  username: string;
  public_ip: string | null;
  ssh_port: number | null;
  winbox_port: number | null;
  model: string | null;
  serial_number: string | null;
  ros_version: string | null;
  board_name: string | null;
  identity: string | null;
  notes: string | null;
  is_active: boolean;
  last_seen: string | null;
  created_at: string;
  updated_at: string;
  created_by: number;
  client_name: string | null;
}

export interface DeviceCreate {
  name: string;
  client_id?: number | null;
  host: string;
  api_port: number;
  use_ssl: boolean;
  username: string;
  password: string;
  public_ip?: string;
  ssh_port?: number;
  winbox_port?: number;
  notes?: string;
}

export interface DeviceUpdate {
  name?: string;
  client_id?: number | null;
  host?: string;
  api_port?: number;
  use_ssl?: boolean;
  username?: string;
  password?: string;
  public_ip?: string;
  ssh_port?: number;
  winbox_port?: number;
  notes?: string;
  is_active?: boolean;
}

export interface DeviceStatus {
  id: number;
  name: string;
  host: string;
  online: boolean;
  uptime: string | null;
  cpu_load: number | null;
  memory_used: number | null;
  memory_total: number | null;
  hdd_used: number | null;
  hdd_total: number | null;
  ros_version: string | null;
  board_name: string | null;
  architecture: string | null;
}
